src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -295,25 +302,41 @@ private static void createImpedantBranch(LfBranch branch, LfBus bus1, LfBus bus2
         }
 
         if (p1 != null) {
-            equationSystem.createEquation(bus1.getNum(), EquationType.BUS_P).addTerm(p1);
+            Equation sp1 = equationSystem.createEquation(bus1.getNum(), EquationType.BUS_P);
+            if (sp1.getTerms().isEmpty()) {
+                bus1.setP(sp1);
+            }
+            sp1.addTerm(p1);
             branch.setP1(p1);
             if (creationParameters.isPhaseControl()) {
                 createBranchActivePowerTargetEquation(branch, DiscretePhaseControl.ControlledSide.ONE, equationSystem, p1);
             }
         }
         if (q1 != null) {
-            equationSystem.createEquation(bus1.getNum(), EquationType.BUS_Q).addTerm(q1);
+            Equation sq1 = equationSystem.createEquation(bus1.getNum(), EquationType.BUS_Q);
+            if (sq1.getTerms().isEmpty()) {
```

Comment:
Not sure I fully understand the purpose of these three lines, is it just a performance optimization, or do they have some deeper implications ?

Reply:
This is just to avoid setting injection equation for each of the branch connected to the bus

Reply:
So I understand that our code only works for networks with at least one branch? I have never tested but can we have a network with only one generator and one load, connected in the same "bus"?